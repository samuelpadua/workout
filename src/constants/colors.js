export default {
  default: '#0086FF',
  success: '#00D604',
  warning: '#FF8A00',
  error: '#FF253A',
  lightGray: '#E5E5E5',
  gray: '#8C8C8C',
  mineShaft: '#404040'
}
