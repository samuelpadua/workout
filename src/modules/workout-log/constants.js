export const WORKOUT_LOG_ADD = 'workoutLog/ADD'
export const WORKOUT_LOG_REMOVE = 'workoutLog/REMOVE'
