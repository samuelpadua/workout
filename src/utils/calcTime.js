function calcTimes(time) {
  const startTime = new Date('1970-1-1 00:00')
  const endTime = new Date(`1970-1-1 ${time}`)
  const diff = (endTime - startTime) / 1000 / 60 / 60

  return diff
}

export default calcTimes
