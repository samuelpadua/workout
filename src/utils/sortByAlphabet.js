const sortByAlphabet = (items = [], orderBy, key) => {
  const asc = (a, b) => {
    if (a[key] > b[key]) return 1
    if (a[key] < b[key]) return -1
    return 0
  }

  const desc = (a, b) => {
    if (a[key] < b[key]) return 1
    if (a[key] > b[key]) return -1
    return 0
  }

  const order = orderBy === 'asc' ? asc : desc

  const ordered = items.sort(order)

  return ordered
}

export default sortByAlphabet
