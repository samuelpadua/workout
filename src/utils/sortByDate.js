const orderByDate = (items = [], orderBy) => {
  const desc = (a, b) => new Date(b.date) - new Date(a.date)
  const asc = (a, b) => new Date(a.date) - new Date(b.date)

  const order = orderBy === 'asc' ? asc : desc

  const ordered = items.sort(order)

  return ordered
}

export default orderByDate
