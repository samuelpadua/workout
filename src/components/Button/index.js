import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { darken } from '../../utils/LightenDarkenColor'
import { colors } from '../../constants'

const ButtonStyled = styled.button`
  width: ${props => (props.full ? '100%' : '')};
  padding: 6px;
  border: 0;
  font: inherit;
  height: 3em;
  border-radius: 4px;
  background: ${props => colors[props.color]};
  box-shadow: none !important;
  outline: 0;
  color: white;

  &:hover {
    background: ${props => darken(colors[props.color], 10)};
  }

  &:active {
    background: ${props => darken(colors[props.color], 20)};
  }
`

function Button(props) {
  const { children } = props

  return (
    <ButtonStyled {...props}>
      { children }
    </ButtonStyled>
  )
}

Button.propTypes = {
  color: PropTypes.oneOf(['default', 'success', 'warning', 'error']),
  full: PropTypes.bool,
  children: PropTypes.node,
}

Button.defaultProps = {
  color: 'default',
  full: false,
  children: '',
}

export default Button
