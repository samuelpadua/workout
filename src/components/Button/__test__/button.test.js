import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import Button from '../index'
import { colors } from '../../../constants'
import { darken } from '../../../utils/LightenDarkenColor'

describe('Button', () => {
  describe('default', () => {
    const tree = renderer
      .create(<Button full>Default</Button>)
      .toJSON()

    it('should render Button', () => {
      expect(tree).toMatchSnapshot()
    })

    it('should change apply dark color when :hover', () => {
      expect(tree).toHaveStyleRule('background', darken(colors.default, 10), {
        modifier: ':hover',
      })
    })

    it('should change apply dark color when :active', () => {
      expect(tree).toHaveStyleRule('background', darken(colors.default, 20), {
        modifier: ':active',
      })
    })
  })

  describe('error', () => {
    const tree = renderer
      .create(<Button color="error" full>Default</Button>)
      .toJSON()

    it('should render Button with color error', () => {
      expect(tree).toMatchSnapshot()
    })

    it('should change apply dark 10% color when :hover', () => {
      expect(tree).toHaveStyleRule('background', darken(colors.error, 10), {
        modifier: ':hover',
      })
    })

    it('should change apply dark 20% color when :active', () => {
      expect(tree).toHaveStyleRule('background', darken(colors.error, 20), {
        modifier: ':active',
      })
    })
  })
})
