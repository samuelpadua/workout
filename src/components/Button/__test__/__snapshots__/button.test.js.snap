// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button default should render Button 1`] = `
.c0 {
  width: 100%;
  padding: 6px;
  border: 0;
  font: inherit;
  height: 3em;
  border-radius: 4px;
  background: #0086FF;
  box-shadow: none !important;
  outline: 0;
  color: white;
}

.c0:hover {
  background: #006de6;
}

.c0:active {
  background: #0053cc;
}

<button
  className="c0"
  color="default"
>
  Default
</button>
`;

exports[`Button error should render Button with color error 1`] = `
.c0 {
  width: 100%;
  padding: 6px;
  border: 0;
  font: inherit;
  height: 3em;
  border-radius: 4px;
  background: #FF253A;
  box-shadow: none !important;
  outline: 0;
  color: white;
}

.c0:hover {
  background: #e60c21;
}

.c0:active {
  background: #cc0007;
}

<button
  className="c0"
  color="error"
>
  Default
</button>
`;
